osquery/events/darwin/endpointsecurity_fim.cpp

```diff
@@ -0,0 +1,196 @@
+      handleMessage(message);
+    };
+
+    auto result = es_new_client(&es_file_client_, handler);
+    if (result == ES_NEW_CLIENT_RESULT_SUCCESS) {
+      es_file_client_success_ = true;
+      return Status::success();
+    } else {
+      return Status::failure(1, getEsNewClientErrorMessage(result));
+    }
+  } else {
+    return Status::failure(
+        1, "EndpointSecurity is only available on macOS 10.15 and higher");
+  }
+}
+
+void EndpointSecurityFileEventPublisher::configure() {
+  if (es_file_client_ == nullptr) {
+    return;
+  }
+
+  auto cache = es_clear_cache(es_file_client_);
+  if (cache != ES_CLEAR_CACHE_RESULT_SUCCESS) {
+    VLOG(1) << "Couldn't clear cache for EndpointSecurity client";
+    return;
+  }
+
+  for (auto& sub : subscriptions_) {
+    auto sc = getSubscriptionContext(sub->context);
+    auto events = sc->es_file_event_subscriptions_;
+
+    for (const auto& p : muted_path_literals_) {
+      auto result = es_mute_path_literal(es_file_client_, p.c_str());
+      if (result == ES_RETURN_ERROR) {
+        VLOG(1) << "Unable to mute path literal: " << p;
+      }
+    }
+
+    for (const auto& p : muted_path_prefixes_) {
+      auto result = es_mute_path_prefix(es_file_client_, p.c_str());
+      if (result == ES_RETURN_ERROR) {
+        VLOG(1) << "Unable to mute path with prefix: " << p;
+      }
+    }
+    auto es_sub = es_subscribe(es_file_client_, &events[0], events.size());
+    if (es_sub != ES_RETURN_SUCCESS) {
+      VLOG(1) << "Couldn't subscribe to EndpointSecurity subsystem";
+    }
+  }
+}
+
+bool EndpointSecurityFileEventPublisher::shouldFire(
+    const EndpointSecurityFileSubscriptionContextRef& sc,
+    const EndpointSecurityFileEventContextRef& ec) const {
+  return true;
+}
+
+void EndpointSecurityFileEventPublisher::tearDown() {
+  if (es_file_client_ == nullptr) {
+    return;
+  }
+  es_unsubscribe_all(es_file_client_);
+  if (es_file_client_success_) {
+    es_delete_client(es_file_client_);
+    es_file_client_ = nullptr;
+  }
+}
+
+void EndpointSecurityFileEventPublisher::handleMessage(
+    const es_message_t* message) {
+  if (message == nullptr) {
+    return;
+  }
+
+  if (message->action_type == ES_ACTION_TYPE_AUTH) {
+    return;
+  }
+  auto ec = createEventContext();
+  ec->version = message->version;
+  if (ec->version >= 2) {
+    ec->seq_num = message->seq_num;
+  }
+
+  if (ec->version >= 4) {
+    ec->global_seq_num = message->global_seq_num;
+  }
+
+  getProperties(message->process, ec);
+
+  switch (message->event_type) {
+  case ES_EVENT_TYPE_NOTIFY_CREATE: {
+    ec->event_type = "create";
+    if (message->event.create.destination_type ==
+        ES_DESTINATION_TYPE_EXISTING_FILE) {
+      ec->filename = getStringFromToken(
+          &message->event.create.destination.existing_file->path);
+    } else {
+      std::string filename = getStringFromToken(
+          &message->event.create.destination.new_path.dir->path);
+      filename += '/';
```

Comment:
consider use boost::filesystem::path to join between paths.

Reply:
Those paths are encapsulated as `es_string_tokens`, and are not direct paths, we just need the string, and doesn't make sense to use boost::path since we don't need the cross platform compatibility